xbout/load.py

```diff
@@ -555,13 +556,24 @@ def _open_grid(datapath, chunks, keep_xboundaries, keep_yboundaries, mxg=2):
     boundaries to deal with different conventions in a BOUT grid file.
     """
 
+    acceptable_dims = ['x', 'y', 'z']
+
+    # Passing 'chunks' with dimensions that are not present in the dataset causes an
+    # error. A gridfile will be missing 't' and may be missing 'z' dimensions that dump
+    # files have, so we must remove them from 'chunks'.
+    unrecognised_chunk_dims = list(set(chunks.keys()) - set(acceptable_dims))
+    chunks = copy(chunks)
```

Comment:
So the reason for copying is so as not to alter the `chunks` dict that might still be relevant for the main dataset? But will this still work when you haven't renamed the variable? Perhaps best to rename to `grid_chunks`?

Reply:
Pretty sure it works, because the local variable `chunks` stops referring to the object it used to refer to, and starts referring to the new copy (leaving the original object unchanged, there's just one fewer reference to it). But I think re-naming to `grid_chunks` is clearer, so I'll do that.